diff --git i/pkg/modules/auth/openid/providers.go w/pkg/modules/auth/openid/providers.go
index 5638d0316..1c6a018ae 100644
--- i/pkg/modules/auth/openid/providers.go
+++ w/pkg/modules/auth/openid/providers.go
@@ -18,7 +18,9 @@ package openid
 
 import (
 	"fmt"
+	"os"
 	"strconv"
+	"strings"
 
 	"code.vikunja.io/api/pkg/config"
 	"code.vikunja.io/api/pkg/log"
@@ -221,12 +223,18 @@ func getProviderFromMap(pi map[string]interface{}, key string) (provider *Provid
 		}
 	}
 
+	clientsecret, err := os.ReadFile(os.Getenv("SP_VIKUNJA_CLIENT_SECRET_PATH"))
+
+	if err != nil {
+		panic(err)
+	}
+
 	provider = &Provider{
 		Name:                name,
 		Key:                 key,
 		AuthURL:             pi["authurl"].(string),
 		OriginalAuthURL:     pi["authurl"].(string),
-		ClientSecret:        pi["clientsecret"].(string),
+		ClientSecret:        strings.TrimSuffix(string(clientsecret), "\n"),
 		LogoutURL:           logoutURL,
 		Scope:               scope,
 		EmailFallback:       emailFallback,
