--- a/pkg/modules/auth/openid/providers.go
+++ b/pkg/modules/auth/openid/providers.go
@@ -19,8 +19,10 @@
 import (
 	"errors"
 	"fmt"
+	"os"
 	"slices"
 	"strconv"
+	"strings"
 
 	"code.vikunja.io/api/pkg/config"
 	"code.vikunja.io/api/pkg/log"
@@ -331,12 +333,17 @@
 		log.Errorf("requireavailability is not a boolean for provider %s: %s", key, err)
 	}
 
+	clientSecret, err := os.ReadFile(os.Getenv("SP_VIKUNJA_CLIENT_SECRET_PATH"))
+	if err != nil {
+		return nil, fmt.Errorf("could not read OpenID client secret: %w", err)
+	}
+
 	provider = &Provider{
 		Name:                name,
 		Key:                 key,
 		AuthURL:             pi["authurl"].(string),
 		OriginalAuthURL:     pi["authurl"].(string),
-		ClientSecret:        pi["clientsecret"].(string),
+		ClientSecret:        strings.TrimSuffix(string(clientSecret), "\n"),
 		LogoutURL:           logoutURL,
 		Scope:               scope,
 		EmailFallback:       emailFallback,
